### PR TITLE
refactor(cli): Use the Cobra framework to enhance CLI

### LINKED ISSUE #87: Merge zbpack and zbplan, make a one entry point
#### Descriptive
Merge `zbpack` and `zbplan`. `zbplan` just print information about user's project, it's important to note that `zbplan` is, in fact, part of `zbpack`.

After Merging, we can use the --info option to output information only.

#### Doing this job has the following benefits:

1. This program has only one entry point, making it not confusing for users.
2. Just release `zbpack` regularly.
3. After each release, we can submit the program to a CLI program package manager, such as `Homebrew`.

#### Points to note:

1. `zbpack` support analyze online Github repo, so when use --info option will support it.
2. Since only show information, check Docker Daemon running status can skip.
3. Update the corresponding documentations.

**I'm willing to do some job to refactor it~**

### PR DESCRIPTION
**Pull request by waymedev**
Close #87 

#### Change log

- move a large of logic from `cmd/zbpack` to `internal/zbpack`

- use `cobra` framework bring default  error message

```
Error: zbpack requires a directory to analyse or build
Usage:
  zbpack [flags]

Flags:
  -h, --help   help for zbpack
  -i, --info   only print project information

```

- use `-h` or `--help` can print help message

```
Zbpack is a powerful tool that not only analyzes your project for dependencies and requirements, but also builds Docker images in one click, greatly simplifying your workflow.

Usage:
  zbpack [flags] <directory path> 

Flags:
  -h, --help   help for zbpack
  -i, --info   only print project information

```

- add some integration test cases but I think it's not enough

**Please feel free to discuss any issues related to code style or other review matters.**

### CODE REVIEW

File: cmd/zbpack/main_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+var binName = "zbpack"
+
+func TestMain(m *testing.M) {
+	fmt.Println("Running build...")
+
+	if runtime.GOOS == "windows" {
+		binName += ".exe"
+	}
+
+	build := exec.Command("go", "build", "-o", binName)
```
Comment (pan93412) [inline cmd/zbpack/main_test.go:22]: Output it to a temporary directory like `/tmp`. Don't place your test executable file in the working directory, which may introduce an unexpected result.

File: cmd/zbpack/main_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			t.Fatal("expected error, but got nil")
+		}
+
+		if !strings.Contains(string(out), "Error") {
+			t.Fatal("expected error output, but got: ", string(out))
+		}
+	})
+
+	t.Run("only show info when give --info flag", func(t *testing.T) {
+		testFilePath := "../../tests/node-corepack"
```
Comment (pan93412) [inline cmd/zbpack/main_test.go:72]: It should be an absolute path with `cmd/zbpack` as base to prevent unexpected result.

File: internal/zbpack/zbpack.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+func plan(path string) error {
+	submoduleName, err := GetSubmoduleName(path)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	githubToken := os.Getenv("GITHUB_ACCESS_TOKEN")
+	if strings.HasPrefix(path, "https://github.com") && githubToken == "" {
+		return fmt.Errorf("GITHUB_ACCESS_TOKEN is required for GitHub URL")
+	}
```
Comment (pan93412) [inline internal/zbpack/zbpack.go:102]: It shouldn't be required IMO. Don't we just clone it?

File: internal/zbpack/zbpack.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+// build is used to build Docker image and show build plan.
+func build(path string) error {
+	if err := CheckDockerDaemonStatus(); err != nil {
+		return err
+	}
+
+	if strings.HasPrefix(path, "https://") || strings.HasPrefix(path, "http://") {
+		return fmt.Errorf("zbpack does not support building from online repositories yet")
+	}
```
Comment (pan93412) [inline internal/zbpack/zbpack.go:65]: hmmmm i think it is possible by cloning the repository to `/tmp` and run a build on that directory; however, it should be written in another PR. At this moment, I recommend you to add a `// TODO` mark.

**Review by pan93412 — CHANGES_REQUESTED** [review]
Good job! However, I found a few issues, and I'm looking forward to your fix ;)

File: internal/zbpack/zbpack.go (hunk 3 above)
Comment (waymedev) [reply internal/zbpack/zbpack.go:102]: The Github token is needed by `PlanOptions`

```
type PlanOptions struct {

	// Access token for GitHub, only used when Path is a GitHub URL.
	AccessToken *string

}
```

I already try remove it but will broken.

As you say, we might refactor it to support building from online repositories.

**Comment by MichaelYuhe** [comment]
LGTM!

**Review by pan93412 — APPROVED** [review]
